server/src/main/java/org/elasticsearch/health/HealthPeriodicLogger.java

```diff
@@ -254,6 +357,49 @@ public void onFailure(Exception e) {
         }
     };
 
+    /**
+     * Handle the setting (and possible creation) of APM metrics
+     */
+    // default visibility for testing purposes
+    void handleMetrics(List<HealthIndicatorResult> healthIndicatorResults) {
+        if (healthIndicatorResults != null) {
+            for (HealthIndicatorResult result : healthIndicatorResults) {
+                String metricName = result.name();
+                LongGaugeMetric metric = this.redMetrics.get(metricName);
+                if (metric == null) {
+                    metric = LongGaugeMetric.create(
+                        this.meterRegistry,
+                        String.format(Locale.ROOT, "es.health.%s.red", metricName),
+                        String.format(Locale.ROOT, "%s: Red", metricName),
+                        "{cluster}"
+                    );
+                    this.redMetrics.put(metricName, metric);
+                }
+                metricWriter.accept(metric, (long) (result.status() == RED ? 1 : 0));
+            }
+
+            metricWriter.accept(this.redMetrics.get("overall"), (long) (calculateOverallStatus(healthIndicatorResults) == RED ? 1 : 0));
+        }
+    }
+
+    private void updateOutputModes(List<OutputMode> newMode) {
+        this.outputModes = newMode;
+    }
+
+    /**
+     * Returns true if any of the outputModes are set to logs
+     */
+    private boolean logsEnabled() {
+        return this.outputModes.stream().anyMatch(m -> m.toString().equals(OutputMode.LOGS.toString()));
```

Comment:
If you make `outputModes` a set, this can be `this.outputModes.contains(OutputMode.LOGS)` instead of the streaming and string converting. Even if you don't, it doesn't need to do the string comparison since it can look at the enum itself